lib/senkyoshi/models/file_resource.rb

```diff
@@ -0,0 +1,23 @@
+require "senkyoshi/models/resource"
+
+module Senkyoshi
+  ##
+  # Class to represent a resource constructed from a single 'dat' file.
+  ##
+  class FileResource < Resource
+    attr_reader(:id)
+
+    def initialize(id = nil)
+      @id = id
+    end
+
+    def self.from(xml = nil, pre_data = nil, _resource_xids = nil)
+      resource = new(pre_data[:file_name])
+      resource.iterate_xml(xml, pre_data)
```

Comment:
Oh my mistake, I thought you were talking about the above method. Each specialization of the FileResource implements some version of `iterate_xml`, so I guess I was using it as a way to communicate that everything that inherits from this method should implement this method to do something more than just return itself. It could probably go though, as it doesn't really do much of anything